### PR TITLE
Update "updateRows" methods to account for new `tr` DOM nodes.

### PR DESCRIPTION
**Pull request by notlee**

### CODE REVIEW

File: src/js/Tables/BaseTable.js [hunk 1]
```diff
@@ -142,11 +142,52 @@ class BaseTable {
...
+	 * Update the o-table instance with rows added or removed dynamically from
+	 * the table. Applies any existing sort and filter to new rows.
 	 *
 	 * @returns {undefined}
 	 */
 	updateRows() {
+		const rows = this._getLatestRowNodes();
+		// Find rows added.
+		const newRows = rows.filter(function (row) {
+			return this.tableRows.indexOf(row) === -1;
```
Comment (JakeChampion) [inline src/js/Tables/BaseTable.js]: Is this to avoid using Array.prototype.includes?

Comment (notlee) [reply src/js/Tables/BaseTable.js]: Yeah

File: src/js/Tables/BaseTable.js [hunk 2]
```diff
@@ -142,11 +142,52 @@ class BaseTable {
...
+		}.bind(this));
```
Comment (JakeChampion) [inline src/js/Tables/BaseTable.js]: instead of doing `.bind(this)` you can pass `this` as the second parameter to `rows.filter`

Comment (notlee) [reply src/js/Tables/BaseTable.js]: fancy pants

Comment (notlee) [reply src/js/Tables/BaseTable.js]: done

File: src/js/Tables/FlatTable.js [hunk 3]
```diff
@@ -14,45 +14,68 @@ class FlatTable extends BaseTable {
...
-				clonedHeader.setAttribute('role', 'rowheader');
-				clonedHeader.classList.add('o-table__duplicate-heading');
-				return clonedHeader;
-			});
-
-			window.requestAnimationFrame(() => {
-				dataHeadings.forEach((clonedHeader, index) => {
-					const td = data[index];
-					td.parentNode.insertBefore(clonedHeader, td);
+	_createFlatTableStructure(rows = this.tableRows) {
```
Comment (rowanmanning) [inline src/js/Tables/FlatTable.js]: I didn't know you can reference `this` in default args!

**Pull request merged**